pkg/extendapi/devicetemplate.go

```diff
@@ -0,0 +1,216 @@
+		return
+	}
+
+	if !gjson.GetBytes(body, "user").Exists() {
+		logrus.Error("CreateDeviceTemplateHealthHandler request body have no field user")
+		w.Write([]byte(`{"error":"request body have no field user"}`))
+		return
+	}
+
+	if !gjson.GetBytes(body, "template").Exists() {
+		logrus.Error("CreateDeviceTemplateHealthHandler request body have no field template")
+		w.Write([]byte(`{"error":"request body have no field template"}`))
+		return
+	}
+
+	deviceType := gjson.GetBytes(body, "deviceType").String()
+	user := gjson.GetBytes(body, "user").String()
+	templateBase64 := gjson.GetBytes(body, "template").String()
+
+	templateBytes, err := base64.StdEncoding.DecodeString(templateBase64)
+	if err != nil {
+		logrus.Error("CreateDeviceTemplateHealthHandler request tmplate base64 decode error:", err.Error)
+		w.Write([]byte(fmt.Sprintf(`{"error":%s}`, err.Error())))
+		return
+	}
+
+	if err := h.validTemplate(req.Context(), deviceType, templateBytes); err != nil {
+		logrus.Error("CreateDeviceTemplateHealthHandler validTemplate error:", err.Error())
+		w.Write([]byte(fmt.Sprintf(`{"error":%s}`, err.Error())))
+		return
+	}
+
+	if err := h.createSecret(req.Context(), deviceType, user, templateBytes); err != nil {
+		logrus.Error("CreateDeviceTemplateHealthHandler createSecret error:", err.Error())
+		w.Write([]byte(fmt.Sprintf(`{"error":%s}`, err.Error())))
+		return
+	}
+
+	w.Write([]byte(fmt.Sprintf(`{"error":success}`)))
+}
+
+func (h *CreateDeviceTemplateHealthHandler) createSecret(ctx context.Context, deviceType, user string, templateBytes []byte) error {
+
+	tempStr := util.GenerateRandomTempKey(7)
+
+	deviceType = strings.ToLower(deviceType)
+
+	name := fmt.Sprintf("%s-template-%s", deviceType, tempStr)
+	stringData := make(map[string]string)
+	stringData["template"] = string(templateBytes)
+	stringData["user"] = user
+
+	secret := &apiv1.Secret{
+		TypeMeta:   metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: "octopus-system", Annotations: map[string]string{"user": user, "deviceType": deviceType}},
+		StringData: stringData,
+	}
+
+	tctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	if _, err := h.clientset.CoreV1().Secrets("octopus-system").Create(tctx, secret, metav1.CreateOptions{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *CreateDeviceTemplateHealthHandler) validTemplate(ctx context.Context, deviceType string, templateBytes []byte) error {
+
+	tempStr := util.GenerateRandomTempKey(7)
+
+	templateMap := make(map[string]interface{})
+
+	js, err := yaml2.YAMLToJSON(templateBytes)
+	if err != nil {
+		return err
+	}
+
+	if err := json.Unmarshal(js, &templateMap); err != nil {
+		return err
+	}
+
+	name := fmt.Sprintf("testdevicelink-%s", tempStr)
+
+	var adaptorName string
+	if deviceType == "BluetoothDevice" {
+		adaptorName = "adaptors.edge.cattle.io/ble"
+	} else {
+		adaptorName = "adaptors.edge.cattle.io/modbus"
+	}
```

Comment:
you should request deviceType from input instead of defining it manually in the code.